agent/resources/bootstrap_configs/relay_bootstrap.yaml

```diff
@@ -65,6 +65,11 @@ static_resources:
           "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
           explicit_http_config:
             http2_protocol_options: {}
+      upstream_connection_options:
+        # Send TCP keep alive in case upstream closing the connection
+        # https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/address.proto#envoy-v3-api-msg-config-core-v3-tcpkeepalive
+        tcp_keepalive:
+          keepalive_time: 200 # The number of seconds a connection needs to be idle before keep-alive probes start being sent.
```

Comment:
What if we just set the `keepalive_interval`? Will it still work?

Or would it fail because keepalive_interval only begins after keepalive_time has reached the linux default of 2 hours?

Reply:
Just setting `keepalive_interval` won't make it work because Relay will only start to send TCP keep alive probes after the default `keepalive_time` 7200s.

`keepalive_interval` means what the interval Envoy should keep sending probes when it hasn't received any reply and when `keepalive_time` has passed and it is time for it to send keep alive.